helpers/mysql/lib/opentelemetry/helpers/mysql.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0module OpenTelemetry
+
+module OpenTelemetry
+  module Helpers
+    # This class contains helpers for MySQL instrumentation libraries, like mysql2 and
+    # trilogy. It is intended for use by instrumentation developers and not to
+    # be called directly by an application.
+    #
+    # To use this in your instrumentation, the `Instrumentation` class for
+    # your gem must contain configuration options for:
+    #  * `:span_name`
+    #  Example:
+    #  `option :span_name, default: :statement_type, validate: %I[statement_type db_name db_operation_and_name]`
+    #
+    # @api public
+    class MySQL
+      QUERY_NAMES = [
+        'set names',
+        'select',
+        'insert',
+        'update',
+        'delete',
+        'begin',
+        'commit',
+        'rollback',
+        'savepoint',
+        'release savepoint',
+        'explain',
+        'drop database',
+        'drop table',
+        'create database',
+        'create table'
+      ].freeze
+
+      QUERY_NAME_REGEX = Regexp.new("^(#{QUERY_NAMES.join('|')})", Regexp::IGNORECASE)
```

Comment:
Regexp are immutable right? I don't recall

Reply:
Nice question! According to Rubocop, they're immutable starting in Ruby 3.0: https://docs.rubocop.org/rubocop/cops_style.html#styleredundantfreeze

I'm happy to remove the freeze in a future PR. It's used here because that's how it was written in the original code.

Reply:
Is there a freeze? I was asking because I didn't see one but thank you for answering my question!